docs/getting-started/java.md

```diff
@@ -0,0 +1,268 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Get Started with the OpenFeature Java SDK and Spring Boot
+
+## Introduction
+
+This walk-through teaches you the basics of using OpenFeature in Java on top of a Spring Boot Web application.
+
+You'll learn how to:
+
+- Integrate OpenFeature Java SDK
+- Install and configure an OpenFeature provider
+- Perform basic feature flagging
+
+## Requirements
+
+This walk-through assumes that:
+
+- You have a basic knowledge of Java and Spring Boot
+- You have Java 8 or later
+- You have Docker installed and running on the host system
+
+## Walk-through
+
+### Step 1: Create a Spring Boot Web application
+
+To get started, visit [Spring Initializer](https://start.spring.io/). This website allows you to download a pre-configured
+Spring boot application. 
+
+For our use case, add **Spring Web** from the dependencies section and set the following **project metadata** values,
+
+- Group: com
+- Artifact: demo
+
+Select a Java version matching your installation and download the application using the **GENERATE** button.
+Extract the archive and open it in your favorite editor.
+
+### Step 2: Add dependencies
+
+Based on the selected build system for Spring Boot application, add OpenFeature Java SDK and flagd provider dependencies
+to existing dependency management configuration. 
+
+Given below are dependencies for Maven and Gradle with the latest dependency versions at the time of writing this walk-through.
+
+<Tabs groupId="dependency">
+<TabItem value="maven" label="Maven">
+
+```xml
+<dependency>
+    <groupId>dev.openfeature</groupId>
+    <artifactId>sdk</artifactId>
+    <version>1.0.0</version>
+</dependency>
+<dependency>
+    <groupId>dev.openfeature.contrib.providers</groupId>
+    <artifactId>flagd</artifactId>
+    <version>0.5.2</version>
+</dependency>
+```
+
+</TabItem>
+<TabItem value="gradle" label="Gradle">
+
+```groovy
+implementation 'dev.openfeature:sdk:1.0.0'
+implementation 'dev.openfeature.contrib.providers:flagd:0.5.1'
+```
+</TabItem>
+</Tabs>
+
+### Step 3: Add code
+
+The `OpenFeatureAPI` class is the main access point to OpenFeature SDK. This class is designed to act as a singleton. Let's define
+a simple bean configuration class called `OpenFeatureBeans`, that allows us to inject the `OpenFeatureAPI` singleton into desired Spring components.
+
+Create the Java class `OpenFeatureBeans.java` inside the package `com.demo` and add the following code,
+```java
+package com.demo;
+
+import dev.openfeature.sdk.OpenFeatureAPI;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class OpenFeatureBeans {
+
+    @Bean
+    public OpenFeatureAPI OpenFeatureAPI() {
+        final OpenFeatureAPI openFeatureAPI = OpenFeatureAPI.getInstance();
+
+        return openFeatureAPI;
+    }
```

Comment:
I think perhaps injecting a client is an improvement over the API itself. The client is typically the thing `application authors` will interact with. The API is useful too, but it's more for one-time top level configuration.

Reply:
My view on `OpenFeatureClient` is that they are specific to usage (ex:- REST endpoint specific, service specific) and is customizable as needed. And `OpenFeatureAPI ` is the singleton with "base configurations" which allows users to generate clients. 

IMO injecting `OpenFeatureAPI ` is the right thing and then users can generate clients on the use case and preserve them for long term use. Let me know if you have different view :)

Reply:
There is also the possibility to expose `Client` singletons from `OpenFeatureBeans` (ex:- HelloOpenFeatureClient with some metadata like name and version), but that could increase the complexity of the guide

Reply:
Both are valid. Client's can be short lived or long, and we can create one per request, even. They are lightweight and generally you can create many. My thinking was just that I would probably want a client for a particular part of the application (package), and the controller(s) could share one for multiple requests since theirs no reason not to here. Additionally, since the API is a singleton that's globally accessible, injecting it as a bean is a bit redundant.

This is just opinion though, I don't mind either way. In a demo application like this either is fine.

Reply:
Thanks for the view and I too agree with your view. But for now, let's keep it and change it later on if we see the need